### PR TITLE
{AKS} `az aks create/update`: Outbound type should not be updated if it is not set

### PR DESCRIPTION
**Pull request by MartinForReal**
**Related command**

**Description**

{AKS}: _get_outbound_type should return None and natgateway profile hould be validaated
**Testing Guide**

**History Notes**

{AKS} `az aks create/update`: Outbound type should not be updated if it is not set

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py [hunk 1]
```diff
@@ -1885,6 +1879,31 @@ def test_get_outbound_type(self):
...
+            self.models,
+            DecoratorMode.CREATE,
+            AgentPoolDecoratorMode.MANAGED_CLUSTER,
+        )
+        ctx_7.attach_agentpool_context(agentpool_ctx_7)
+        # fail on mutually exclusive outbound_type and managed_outbound_ipv6_count on
+        # load balancer
+        # fail on invalid load_balancer_sku (basic) when outbound_type is CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING
+        with self.assertRaises(InvalidArgumentValueError):
+            ctx_2.get_outbound_type()
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py:1906]:
```suggestion
            ctx_7.get_outbound_type()
```

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py [hunk 2]
```diff
@@ -1885,6 +1879,31 @@ def test_get_outbound_type(self):
...
+            self.cmd,
+            AKSAgentPoolParamDict({"vnet_subnet_id": "test_vnet_subnet_id"}),
+            self.models,
+            DecoratorMode.CREATE,
+            AgentPoolDecoratorMode.MANAGED_CLUSTER,
+        )
+        ctx_7.attach_agentpool_context(agentpool_ctx_7)
+        # fail on mutually exclusive outbound_type and managed_outbound_ipv6_count on
+        # load balancer
+        # fail on invalid load_balancer_sku (basic) when outbound_type is CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py:1904]: fail on mutual exclusion between UDR outbound and custom managed NAT setting?

**Review by FumingZhang — COMMENTED** [review]
Can you add a case to verify that the situation reported by the customer can now be properly handled? (update scenario, value of outbound type is not camel cased, should pass cli validation)

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py (hunk 1 above)
Comment (MartinForReal) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py]: it is fixed. Thanks!

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py (hunk 2 above)
Comment (MartinForReal) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py]: fixed. Thanks!

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py [hunk 3]
```diff
@@ -1885,6 +1879,48 @@ def test_get_outbound_type(self):
...
+            AKSManagedClusterParamDict(
+                {
+                }
+            ),
+            self.models,
+            DecoratorMode.UPDATE,
+        )
+        ctx_8.attach_mc(mc)
+        # fail on mutually exclusive outbound_type and nat_gateway_managed_outbound_ip_count on
+        # nat gateway
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py:1921]: nothing failed, right?

**Review by FumingZhang — COMMENTED** [review]
lgtm

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py (hunk 3 above)
Comment (MartinForReal) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py]: fixed.Thanks

**Review by FumingZhang — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**